crates/time_format/src/time_format.rs

```diff
@@ -0,0 +1,303 @@
+        return formatted_time;
+    }
+
+    if reference_local_date.previous_day() == Some(timestamp_local_date) {
+        return format!("yesterday at {}", formatted_time);
+    }
+
+    match meridiem {
+        Some(_) => format!(
+            "{:02}/{:02}/{}",
+            timestamp_local_date.month() as u32,
+            timestamp_local_date.day(),
+            timestamp_local_date.year()
+        ),
+        None => format!(
+            "{:02}/{:02}/{}",
+            timestamp_local_date.day(),
+            timestamp_local_date.month() as u32,
+            timestamp_local_date.year()
+        ),
+    }
+}
+
+/// Returns true if the locale is recognized as a 12-hour time locale.
+fn is_12_hour_time_by_locale(locale: &str) -> bool {
+    [
+        "es-MX", "es-CO", "es-SV", "es-NI",
+        "es-HN", // Mexico, Colombia, El Salvador, Nicaragua, Honduras
+        "en-US", "en-CA", "en-AU", "en-NZ", // U.S, Canada, Australia, New Zealand
+        "ar-SA", "ar-EG", "ar-JO", // Saudi Arabia, Egypt, Jordan
+        "en-IN", "hi-IN", // India, Hindu
+        "en-PK", "ur-PK", // Pakistan, Urdu
+        "en-PH", "fil-PH", // Philippines, Filipino
+        "bn-BD", "ccp-BD", // Bangladesh, Chakma
+        "en-IE", "ga-IE", // Ireland, Irish
+        "en-MY", "ms-MY", // Malaysia, Malay
+    ]
+    .contains(&locale)
+}
+
+#[cfg(target_os = "macos")]
+mod macos {
+    use super::*;
+    use core_foundation::base::TCFType;
+    use core_foundation::date::CFAbsoluteTime;
+    use core_foundation::string::CFString;
+    use core_foundation_sys::date_formatter::CFDateFormatterCreateStringWithAbsoluteTime;
+    use core_foundation_sys::date_formatter::CFDateFormatterRef;
+    use core_foundation_sys::locale::CFLocaleRef;
+    use core_foundation_sys::{
+        base::kCFAllocatorDefault,
+        date_formatter::{
+            kCFDateFormatterNoStyle, kCFDateFormatterShortStyle, CFDateFormatterCreate,
+        },
+        locale::CFLocaleCopyCurrent,
+    };
+
+    pub fn format_time(timestamp: &time::OffsetDateTime) -> Result<String> {
+        format_with_date_formatter(timestamp, TIME_FORMATTER.with(|f| *f))
+    }
+
+    pub fn format_date(timestamp: &time::OffsetDateTime) -> Result<String> {
+        format_with_date_formatter(timestamp, DATE_FORMATTER.with(|f| *f))
+    }
+
+    fn format_with_date_formatter(
+        timestamp: &time::OffsetDateTime,
+        fmt: CFDateFormatterRef,
+    ) -> Result<String> {
+        const UNIX_TO_CF_ABSOLUTE_TIME_OFFSET: i64 = 978307200;
+        // Convert timestamp to macOS absolute time
+        let timestamp_macos = timestamp.unix_timestamp() - UNIX_TO_CF_ABSOLUTE_TIME_OFFSET;
+        let cf_absolute_time = timestamp_macos as CFAbsoluteTime;
+        unsafe {
+            let s = CFDateFormatterCreateStringWithAbsoluteTime(
+                kCFAllocatorDefault,
+                fmt,
+                cf_absolute_time,
+            );
+            Ok(CFString::wrap_under_create_rule(s).to_string())
```

Comment:
My understanding is that CFString will be released as soon as we drop the rust `String` which is what we are returning here. So there should be no leak, is that correct?